html/cross-origin-opener-policy/resources/common.js

```diff
@@ -0,0 +1,36 @@
+const SAME_ORIGIN = {origin: get_host_info().HTTPS_ORIGIN, name: "SAME_ORIGIN"};
+const SAME_SITE = {origin: get_host_info().HTTPS_REMOTE_ORIGIN, name: "SAME_SITE"};
+const CROSS_ORIGIN = {origin: get_host_info().HTTPS_NOTSAMESITE_ORIGIN, name: "CROSS_ORIGIN"}
+
+function url_test(t, url, channelName, hasOpener) {
+  const bc = new BroadcastChannel(channelName);
+  bc.onmessage = t.step_func_done(event => {
+    const payload = event.data;
+    assert_equals(payload.name, hasOpener ? channelName : "");
+    assert_equals(payload.opener, hasOpener);
+  });
+
+  const w = window.open(url, channelName);
+
+  // w will be closed by its postback iframe. When out of process,
+  // window.close() does not work.
+  t.add_cleanup(() => w.close());
+}
+
+function coop_coep_test(t, host, coop, coep, channelName, hasOpener) {
+  url_test(t, `${host.origin}/html/cross-origin-opener-policy/resources/coop-coep.py?coop=${encodeURIComponent(coop)}&coep=${coep}&channel=${channelName}`, channelName, hasOpener);
```

Comment:
This is too long. One solution might be
let url = `${host.origin}/html/cross-origin-opener-policy/resources/coop-coep.py`;
url += `?coop=...